src/DotNetLightning.Core/Channel/CommitmentToLocalExtension.fs

```diff
@@ -81,47 +81,28 @@ type CommitmentToLocalParameters = {
 
 type internal CommitmentToLocalExtension() =
     inherit BuilderExtension()
-        override __.CanGenerateScriptSig (scriptPubKey: Script): bool =
-            (CommitmentToLocalParameters.TryExtractParameters scriptPubKey).IsSome
+        override __.Match (coin: ICoin, _input: PSBTInput): bool =
+            (CommitmentToLocalParameters.TryExtractParameters (coin.GetScriptCode())).IsSome
 
-        override __.GenerateScriptSig(scriptPubKey: Script, keyRepo: IKeyRepository, signer: ISigner): Script =
-            let parameters =
-                match (CommitmentToLocalParameters.TryExtractParameters scriptPubKey) with
-                | Some parameters -> parameters
-                | None ->
-                    failwith
-                        "NBitcoin should not call this unless CanGenerateScriptSig returns true"
-            let pubKey = keyRepo.FindKey scriptPubKey
-            // FindKey will return null if it can't find a key for
-            // scriptPubKey. If we can't find a valid key then this method
-            // should return null, indicating to NBitcoin that the sigScript
-            // could not be generated.
-            match pubKey with
-            | null -> null
-            | _ when pubKey = parameters.RevocationPubKey.RawPubKey() ->
-                let revocationSig = signer.Sign (parameters.RevocationPubKey.RawPubKey())
-                Script [
-                    Op.GetPushOp (revocationSig.ToBytes())
-                    Op.op_Implicit OpcodeType.OP_TRUE
-                ]
-            | _ when pubKey = parameters.LocalDelayedPubKey.RawPubKey() ->
-                let localDelayedSig = signer.Sign (parameters.LocalDelayedPubKey.RawPubKey())
-                Script [
-                    Op.GetPushOp (localDelayedSig.ToBytes())
-                    Op.op_Implicit OpcodeType.OP_FALSE
-                ]
-            | _ -> null
+        override __.Sign(inputSigningContext: InputSigningContext, keyRepo: IKeyRepository, signer: ISigner) =
+            let scriptPubKey = inputSigningContext.Coin.GetScriptCode()
+
+            let pubKey = keyRepo.FindKey scriptPubKey :?> PubKey
```

Comment:
Wny don't match against type like this 
```fsharp
match keyRepo.FindKey scriptPubKey with
| :? PubKey as pubkey when pubkey |> isNull |> not ->
   // ... 
```

Which is more type safe

Reply:
actually, it should use TryFind with is more F# idiomatic, right?

Reply:
> actually, it should use TryFind which is more F# idiomatic, right?

As far as I can see, IKeyRepository doesn't have any other member except FindKey

Reply:
> Wny don't match against type like this
> 
> ```fsharp
> match keyRepo.FindKey scriptPubKey with
> | :? PubKey as pubkey when pubkey |> isNull |> not ->
>    // ... 
> ```
> 
> Which is more type safe

Done